optional/vtd/vendor/vtd-2022.3/conanfile.py

```diff
@@ -0,0 +1,231 @@
+    subprocess.run(
+        ["patchelf", "--set-rpath", rpath_arg, file_arg],
+        check=True,
+    )
+
+
+def find_binary_files() -> List[Path]:
+    """Return a list of all file paths that are of the binary type."""
+    result = subprocess.run(
+        """find -type f -executable -exec sh -c "file -i '{}' | grep -q '; charset=binary'" \; -print""",
+        shell=True,
+        stdout=subprocess.PIPE,
+        check=True,
+    )
+    return [Path(x) for x in result.stdout.decode().splitlines()]
+
+def remove_broken_symlinks() -> None:
+    """Remove all broken symlinks that are in the vtd_folder."""
+    subprocess.run(
+        """rm $(find . -xtype l)""",
+        shell=True,
+        stdout=subprocess.PIPE,
+        check=True,
+    )
+
+
+class VtdConan(ConanFile):
+    name = "vtd"
+    version = "2022.3"
+    license = "Proprietary"
+    url = "https://vires.mscsoftware.com"
+    no_copy_source = True
+    description = "Include binary files in C/C++"
+    topics = ("simulator", "vires")
+    settings = "os", "arch"
+    options = {
+        "with_osi": [True, False],
+        "with_road_designer": [True, False],
+        "with_image_generator":  [True, False], 
+    }
+    default_options = {
+        "with_osi": True,
+        "with_road_designer": False,
+        "with_image_generator": False,
+    }
+
+    _archive_base = "dl/VTD.2022.3.tgz"
+    _archive_osi = "dl/vtd.2020.addOns.OSI.20200826.tgz"
+    _archive_rod = "dl/vtd.2021.1.addOns.World.Edit.Base.20210511.tgz"
+    _root_dir = "VTD.2022.3"
+
+    def export_sources(self):
+        self.copy("libdeps_pruned.txt")
+        self.copy(self._archive_base, symlinks=False)
+        if Path(self._archive_osi).exists():
+            self.copy(self._archive_osi, symlinks=False)
+        if Path(self._archive_rod).exists():
+            self.copy(self._archive_rod, symlinks=False)
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("VTD binaries do not exist for Windows")
+
+    def build(self):
+        src = Path(self.source_folder)
+        dst = Path(self.build_folder)
+        vtddir = dst / "VTD.2022.3"
+        libdir = vtddir / "bundled"
+
+        def extract_archive(archive):
+            print(f"Extracting: {archive}")
+            tools.untargz(src / archive, dst)
+        
+        extract_archive(self._archive_base)
+        libdir.mkdir()
+        if self.options.with_osi:
+            extract_archive(self._archive_osi)
+        if self.options.with_road_designer:
+            extract_archive(self._archive_rod)
+        if not self.options.with_image_generator:
+            path = Path(vtddir / "Runtime/Core/IG64").resolve()
+            shutil.rmtree(path)
+            Path(vtddir / "Runtime/Core/ImageGenerator").unlink()
+            remove_broken_symlinks()
+            
+        # Patch RPATH of several critical binaries.
+        patch_rpath(
```

Comment:
This section where stuff is patched could be formatted nicer.
Each `patch_rpath` invocation a single line, for example.